src/Microsoft.Azure.ServiceBus/Primitives/AzureActiveDirectoryTokenProvider.cs

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.Azure.ServiceBus.Primitives
+{
+    using System;
+    using System.Threading.Tasks;
+    using Microsoft.IdentityModel.Clients.ActiveDirectory;
+
+    /// <summary>
+    /// Represents the Azure Active Directory token provider for the Service Bus.
+    /// </summary>
+    public class AzureActiveDirectoryTokenProvider : TokenProvider
```

Comment:
API needs to be re-approved to show public types that will be exposed as part of this PR by running 
https://github.com/Azure/azure-service-bus-dotnet/blob/560ed609c9d491cc77e05018a630cc169e176aa4/test/Microsoft.Azure.ServiceBus.UnitTests/API/APIApprovals.cs#L15